### PR TITLE
Borders: Remove unused BorderStyleControl component

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
## What?

Removes the `BorderStyleControl` component from the block editor.

## Why?

The `BorderStyleControl` component is no longer in use and has been superceded by the new `BorderControl` and `BorderBoxControl` components which include a border style picker. 

## How?

Given the component was experimental, it has simply been removed. 

## Testing Instructions

1. Search code for any missed references for the component etc

### CODE REVIEW

**Comment by glendaviesnz** [comment]
Looks good to me, I couldn't see any other uses and everything built and ran ok, but I would be interested to hear from someone that has had experience with removing experimental  APIs before signing off on it, just to confirm no deprecation period is needed.

**Comment by aaronrobertshaw** [comment]
@glendaviesnz I've asked around with those more in the know and was given the following breakdown of our deprecation strategy.

- If it is `__unstable` or `__experimental`, we can simply delete it
- If it is only in the plugin, we need to `deprecate()` then delete after 2 releases
- If it is in Core, we need to maintain support for it

The first point above is also supported by the [Coding Guidelines](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/coding-guidelines.md#experimental-and-unstable-apis)

So for this PR, we can delete this without any deprecation steps. I've updated the PR description to remove that question.

Are you happy to approve this PR on that advice or shall we wait on a separate approval?

cc/ @noisysocks

**Review by talldan — APPROVED** [review]
Looks good 👍

**Pull request merged**
